src/stream/src/executor/watermark_filter.rs

```diff
@@ -242,39 +247,59 @@ impl<S: StateStore> WatermarkFilterExecutor<S> {
                     }
 
                     table.commit(barrier.epoch).await?;
+                    yield Message::Barrier(barrier);
+
+                    if need_update_global_max_watermark {
+                        current_watermark = Self::get_global_max_watermark(
+                            &table,
+                            &global_watermark_table,
+                            HummockReadEpoch::Committed(prev_epoch),
```

Comment:
I guess `NoWait` is still okay here, cuz we're sure `prev_epoch` of the `Pause` checkpoint is committed here.

https://github.com/risingwavelabs/risingwave/blob/46057734a9bf49631f56c466fc8ade5c239530f1/src/meta/src/barrier/command.rs#L892-L900